### PR TITLE
[PW-3431] Merge generic component to 7.0.0-rc.1

### PR DESCRIPTION
**Pull request by AlexandrosMor**
**Description**

**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

**Review by msilvagarcia — APPROVED** [review]

File: Helper/PaymentMethods.php [hunk 1]
```diff
@@ -398,38 +307,115 @@ protected function getCurrentShopperReference()
...
-        // initialize service
-        $service = $this->adyenHelper->createAdyenPosPaymentService($client);
+        if (!empty($this->getCurrentShopperReference())) {
+            $paymentMethodRequest["shopperReference"] = $this->getCurrentShopperReference();
+        }
 
-        $requestParams = [
-            "merchantAccount" => $this->adyenHelper->getAdyenMerchantAccount('adyen_pos_cloud', $storeId),
-        ];
+        $amountValue = $this->adyenHelper->formatAmount($this->getCurrentPaymentAmount(), $currencyCode);
```
Comment (cyattilakiss) [inline Helper/PaymentMethods.php:339]: In this brach we already use the latest library where the currency sanitize function is available to get the minor units for an amount.

The adyenHelper->formatAmount can be replaced with the function from the library and the helper functions can be removed from the adyenHelper like: https://github.com/Adyen/adyen-magento2/blob/7.0.0-rc.1/Helper/Data.php#L279-L312 and https://github.com/Adyen/adyen-magento2/blob/7.0.0-rc.1/Helper/Data.php#L321-L324

File: Helper/PaymentMethods.php [hunk 2]
```diff
@@ -398,38 +307,115 @@ protected function getCurrentShopperReference()
...
+                        'width' => 77,
+                        'height' => 50
+                    ];
+                }
+
+                $paymentMethodsExtraDetails[$paymentMethodCode]['icon'] = $icon;
+
+                //todo check if it is needed
+                // check if payment method is an open invoice method
+                $paymentMethodsExtraDetails[$paymentMethodCode]['isOpenInvoice'] =
```
Comment (cyattilakiss) [inline Helper/PaymentMethods.php:415]: create tickets for these todos

File: Helper/Requests.php [hunk 3]
```diff
@@ -327,6 +300,7 @@ public function buildBrowserData($request = [])
 
     /**
      * @param array $request
+<<<<<<< HEAD
```
Comment (cyattilakiss) [inline Helper/Requests.php:303]:
```suggestion
```

File: Helper/PaymentMethods.php (hunk 1 above)
Comment (AlexandrosMor) [reply Helper/PaymentMethods.php:339]: Ticket created PW-3512

**Review by cyattilakiss — APPROVED** [review]

**Pull request merged**